tests/commerce_integration/TEC/Gateways/Stripe/Webhooks/PaymentIntentWebhookTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace TEC\Gateways\Stripe\Webhooks;
+
+use TEC\Tickets\Commerce\Gateways\Stripe\Webhooks\Payment_Intent_Webhook;
+use TEC\Tickets\Commerce\Gateways\Stripe\Status;
+use TEC\Tickets\Commerce\Status\Pending;
+use TEC\Tickets\Commerce\Status\Action_Required;
+use TEC\Tickets\Commerce\Status\Completed;
+use TEC\Tickets\Commerce\Status\Refunded;
+
+class PaymentIntentWebhookTest extends \Codeception\TestCase\WPTestCase {
+	/**
+	 * @dataProvider data_provider_should_payment_intent_be_updated
+	 */
+	public function test_should_payment_intent_be_updated( array $payment_intent_received, array $payment_intents_stored, bool $expected ) {
+		$actual = Payment_Intent_Webhook::should_payment_intent_be_updated( $payment_intent_received, $payment_intents_stored );
+		$this->assertEquals( $expected, $actual );
+	}
+
+	public function data_provider_should_payment_intent_be_updated() {
+		return [
+			// Scenario 1: Payment intent was reset or processing restarted
+			[
+				['status' => Status::REQUIRES_PAYMENT_METHOD],
+				[
+					Status::REQUIRES_PAYMENT_METHOD => [['id' => 'pi_1']],
+					Status::SUCCEEDED => [['id' => 'pi_2']]
```

Comment:
Why does this one change rom `pi_1` to `pi_2`?

Reply:
I was just adding an extra variable in there. In this scenario, the only thing it should be looking at is the `Status::REQUIRES_PAYMENT_METHOD` and the fact that there are more than 1 in the array of the 2nd parameter. The IDs shouldn't matter.